CoseHandler/CoseHandler.cs

```diff
@@ -591,27 +592,46 @@ public static class CoseHandler
         if (publicKey is null)
         {
             errorCodes.Add(ValidationFailureCode.NoPublicKey);
-            return new ValidationResult(false, errorCodes, certValidationResults, chain);
+            return new ValidationResult(false, errorCodes, certValidationResults, chain, validationType: ContentValidationType.ContentValidationNotPerformed);
         }
 
         // Validate that the COSE header is formatted correctly and that the payload and hash are consistent.
         bool messageVerified = false;
+
+        // check for external payload
+        bool hasBytes = !payloadBytes.IsNullOrEmpty();
+        bool hasStream = payloadStream is not null;
+
+        // Check for an indirect signature, where the content header contains the hash of the payload, and the algorithm is stored in the message.
+        // If this is the case and external content is provided, we can validate an external payload hash against the hash stored in the cose message content.
+        bool indirectSignature = msg.IsDetachedSignature();

```

Comment:
You probably don't need to set this anymore, if it's only used in one place. And you might refactor the comments above it into the new logic flow.